lib/config.py

```diff
@@ -155,47 +46,61 @@ def __init__(self):
         # create the top-level parser
         self.parser = argparse.ArgumentParser()
         self._CONF = None
-        self._setup_command_line_parser(SUBCOMMANDS)
+        self._setup_command_line_parser(CONFIG_METADATA_FILE_PATH)
 
     @property
     def CONF(self):
         return self._CONF
 
 
-    def _setup_command_line_parser(self, subcommands):
+    def _setup_command_line_parser(self, config_metadata_file_path):
         """
         Configures the argument parser object to match the expected
             configuration.
 
         Args:
-            subcommands ([(str, str, [dict])]): subcommands. Each subcommand is
-                a tuple with subcommand name, subcommand help message, subcommand arguments.
-                Each subcommand argument is a dict from a tuple with argument possible names
-                to a dict with argument attributes. See ArgParse documentation for possible
-                values in this dict.
+            config_metadata_file_path (str): config metadata file path
         """
         self.parser.add_argument('--config-file', '-c',
                                  help='Path of the configuration file for build '
                                       'scripts',
                                  # NOTE(maurosr): move this to /etc in the future
                                  default='./config.yaml')
-        self.parser.add_argument('--log-file', '-l',
-                                 help='Log file')
-        self.parser.add_argument('--verbose', '-v',
-                                 help='Set the scripts to be verbose',
-                                 action='store_true')
-        self.parser.add_argument('--log-size',
-                                 help='Size in bytes above which the log file '
-                                 'should rotate', type=int)
 
         subparsers = self.parser.add_subparsers(
             dest="subcommand",
             help="Available subcommands")
-        for command, help_msg, arg_groups in subcommands:
-            command_parser = subparsers.add_parser(command, help=help_msg)
-            for arg_group in arg_groups:
-                for arg, options in arg_group.items():
-                    command_parser.add_argument(*arg, **options)
+
+        # setup parser from config metadata file
```

Comment:
First of all, why do we need to infer type, nargs, action? Wasn't that the supposition when we *didn't* have a config_metadata file? Why not simply keep these explicitly as well as any other `add_argument` argument? I think it would simplify the code below, avoid any bugs in the inferring logic and also allow us to use other arguments.

Or am I forgetting something here?